### ISSUE TITLE
Login Success despite canceling sign in

### ISSUE THREAD

**Issue by sam-mckay** [playgameservices/play-games-plugin-for-unity#2632]
If you have a google play games account that is setup to automatically sign you in to games but hasn't used a particular game before you get shown a dialog asking for permission to connect to that game "Allow {gameName} access to your data" if instead you press the android back button the dialog is closed but the user is still logged into the game

For our game in particular this breaks login flow since we do not expect a user who is both authenticated and cancelled the login process (i.e. didn't log in). 

The bigger concern here is a privacy one... the user hasn't given access to their data but has still given their data.  

Tested on 0.9.62 with unity 2017.4.29f1

**Comment by sam-mckay** [comment]
Repro'd via a completely fresh emulator, added two accounts, one of which is brand new and never accepted the data policy popup, set that one as automatic sign in, keep hitting back button on start up, observe you get logged in

![gpgsloginbug](https://user-images.githubusercontent.com/9663483/61961566-8bc3f380-afbf-11e9-892a-8f8bcc7feb43.gif)

Below code taken from the README (with irrelevant config settings removed)
```
    void Start()
    {
        PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder()
        // enables saving game progress.        
        // requests the email address of the player be available.
        // Will bring up a prompt for consent.
        .RequestEmail()
        // requests a server auth code be generated so it can be passed to an
        //  associated back end server application and exchanged for an OAuth token.
        .RequestServerAuthCode(false)
        // requests an ID token be generated.  This OAuth token can be used to
        //  identify the player to other services such as Firebase.
        .RequestIdToken()
        .Build();


        PlayGamesPlatform.InitializeInstance(config);
        PlayGamesPlatform.DebugLogEnabled = true;
        PlayGamesPlatform.Activate();
        Social.localUser.Authenticate((bool success) =>
        {
        });
    }
```

**Comment by olehkuznetsov** [comment]
Should be fixed in v0.10.06. Feel free to reopen if it reproduces there.
